### PR TITLE
Improves descriptions of Integrated Circuits

### PR DESCRIPTION
**Pull request by Alexch2**
[Changelogs]: # (Your PR should contain a detailed changelog of notable changes, titled and categorized appropriately. This includes, new features, sprites, sounds, balance changes, admin tools, map edits, removals, big refactors, config changes, hosting changes and important fixes. An example changelog has been provided below for you to edit. If you need additional help, read https://github.com/tgstation/tgstation/wiki/Changelogs)

:cl: Alexch2
spellcheck: Fixes tons of typos related to integrated circuits.
tweak: Re-writes the descriptions of a few parts of integrated circuits.
/:cl:

[why]: # (Please add a short description [two lines down] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.)

Many items in the integrated circuit printers had very poor, and sometimes downright confusing descriptions. This fixes that.

### CODE REVIEW

**Comment by arsserpentarium** [comment]
please, if you rename circuits, make subtype with right name and remove spawnlags from old name.To avoid blueprints problems.

**Comment by Alexch2** [comment]
Sorry, I forgot about the backwards compatibility stuff on circuits. I reverted the name-change on the coordinate pathfinder back to coordinte.

File: code/modules/integrated_electronics/subtypes/manipulation.dm [hunk 1]
```diff
@@ -410,7 +410,7 @@
 	name = "pulling claw"
 	desc = "Circuit which can pull things.."
 	icon_state = "pull_claw"
-	extended_desc = "The circuit accepts a reference to thing to be pulled. Modes: 0 for release. 1 for pull."
+	extended_desc = "This circuit accepts a reference to the thing to be pulled. Modes: 0 for release. 1 for pull."
```
Comment (PKPenguin321) [inline code/modules/integrated_electronics/subtypes/manipulation.dm]: `a`, not `the `

File: code/modules/integrated_electronics/subtypes/output.dm [hunk 2]
```diff
@@ -157,7 +157,7 @@
 
 /obj/item/integrated_circuit/output/sound/beeper
 	name = "beeper circuit"
-	desc = "Takes a sound name as an input, and will play said sound when pulsed. This circuit has a variety of beeps, boops, and buzzes to choose from."
+	desc = "Takes a sound name as an input, and will play that sound when pulsed. This circuit has a variety of beeps, boops, and buzzes to choose from."
```
Comment (PKPenguin321) [inline code/modules/integrated_electronics/subtypes/output.dm]: Unneeded, it was already correct

File: code/modules/integrated_electronics/subtypes/output.dm [hunk 3]
```diff
@@ -172,7 +172,7 @@
 
 /obj/item/integrated_circuit/output/sound/beepsky
 	name = "securitron sound circuit"
-	desc = "Takes a sound name as an input, and will play said sound when pulsed. This circuit is similar to those used in Securitrons."
+	desc = "Takes a sound name as an input, and will play that sound when pulsed. This circuit is similar to those used in Securitrons."
```
Comment (PKPenguin321) [inline code/modules/integrated_electronics/subtypes/output.dm]: Ditto

File: code/modules/integrated_electronics/subtypes/output.dm [hunk 4]
```diff
@@ -187,7 +187,7 @@
 
 /obj/item/integrated_circuit/output/sound/medbot
 	name = "medbot sound circuit"
-	desc = "Takes a sound name as an input, and will play said sound when pulsed. This circuit is often found in medical robots."
+	desc = "Takes a sound name as an input, and will play that sound when pulsed. This circuit is often found in medical robots."
```
Comment (PKPenguin321) [inline code/modules/integrated_electronics/subtypes/output.dm]: Ditto

File: code/modules/integrated_electronics/subtypes/output.dm [hunk 5]
```diff
@@ -209,7 +209,7 @@
 
 /obj/item/integrated_circuit/output/sound/vox
 	name = "ai vox sound circuit"
-	desc = "Takes a sound name as an input, and will play said sound when pulsed. This circuit is often found in AI announcement systems."
+	desc = "Takes a sound name as an input, and will play that sound when pulsed. This circuit is often found in AI announcement systems."
```
Comment (PKPenguin321) [inline code/modules/integrated_electronics/subtypes/output.dm]: Ditto

File: code/modules/integrated_electronics/subtypes/reagents.dm [hunk 6]
```diff
@@ -21,7 +21,7 @@
 	desc = "Unlike most electronics, creating smoke is completely intentional."
 	icon_state = "smoke"
 	extended_desc = "This smoke generator creates clouds of smoke on command. It can also hold liquids inside, which will go \
-	into the smoke clouds when activated. The reagents are consumed when smoke is made."
+	into the smoke clouds when activated. The reagents are consumed when the smoke is made."
```
Comment (PKPenguin321) [inline code/modules/integrated_electronics/subtypes/reagents.dm]: Unneeded, but either is fine

File: code/modules/integrated_electronics/subtypes/reagents.dm [hunk 7]
```diff
@@ -99,10 +99,10 @@
 
 /obj/item/integrated_circuit/reagent/injector
 	name = "integrated hypo-injector"
-	desc = "This scary looking thing is able to pump liquids into whatever it's pointed at."
+	desc = "This scary looking thing is able to pump liquids into, or suck liquids out of whatever it's pointed at."
```
Comment (PKPenguin321) [inline code/modules/integrated_electronics/subtypes/reagents.dm]: Comma after `out of`

File: code/modules/integrated_electronics/subtypes/smart.dm [hunk 8]
```diff
@@ -74,8 +75,8 @@
 /obj/item/integrated_circuit/smart/advanced_pathfinder
 	name = "advanced pathfinder"
 	desc = "This circuit uses a complex processor for long-range pathfinding."
-	extended_desc = "This circuit uses absolute coordinates for target. A path will be generated taking obstacle input into account, \
-	pathing around any instances of said input. The passkey provided from a card reader is used to create a valid path through doorways."
+	extended_desc = "This circuit uses absolute coordinates for its target. A path will be generated to the target, taking the obstacle input into account, \
```
Comment (PKPenguin321) [inline code/modules/integrated_electronics/subtypes/smart.dm]: uses absolute coordinates *to find* its target

taking *obstacles* into account

**Review by PKPenguin321 — CHANGES_REQUESTED** [review]
Some changes actually make the grammar worse, some aren't necessary, and some are incomplete

**Comment by Alexch2** [comment]
Yikes, that's embarrassing! I'll fix, triple-check, and reopen the PR later today.

File: code/modules/integrated_electronics/subtypes/data_transfer.dm [hunk 9]
```diff
@@ -4,7 +4,7 @@
 
 /obj/item/integrated_circuit/transfer/multiplexer
 	name = "two multiplexer"
-	desc = "This is what those in the business tend to refer to as a 'mux' or data selector. It moves data from one of the selected inputs to the output."
+	desc = "This is what those in the business tend to refer to as a 'mux', or data selector. It moves data from one of the selected inputs to the output."
```
[the remark this answers is not in the document]
Comment (Alexch2) [reply code/modules/integrated_electronics/subtypes/data_transfer.dm]: Isn't "commas inside or outside quotes" just a question of British or American English? Is the standard for tgstation code American English?

File: code/modules/integrated_electronics/subtypes/input.dm [hunk 10]
```diff
@@ -358,9 +358,9 @@
 
 /obj/item/integrated_circuit/input/turfpoint
 	name = "Tile pointer"
```
[the remark this answers is not in the document]
Comment (Alexch2) [reply code/modules/integrated_electronics/subtypes/input.dm]: I'm not sure if analysed circuit code is case sensitive or not, so I'm unsure if changing the name is a good idea.

File: code/modules/integrated_electronics/subtypes/data_transfer.dm (hunk 9 above)
[the remark this answers is not in the document]
Comment (PKPenguin321) [reply code/modules/integrated_electronics/subtypes/data_transfer.dm]: I actually don't know, I'm really only familiar with American grammar rules. The codebase doesn't actually have a standard, so if this is really the case then I guess whichever is fine.

File: code/modules/integrated_electronics/subtypes/input.dm (hunk 10 above)
[the remark this answers is not in the document]
Comment (PKPenguin321) [reply code/modules/integrated_electronics/subtypes/input.dm]: Good point. If you feel like testing it go ahead, but you don't need to do this.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by Alexch2** [comment]
@ShizCalev Done, since several maintainers are telling me to do it.

File: code/modules/integrated_electronics/subtypes/access.dm [hunk 11]
```diff
@@ -1,6 +1,6 @@
 /obj/item/integrated_circuit/input/card_reader
 	name = "card reader"
-	desc = "A circuit that can read registred name, assignment and a PassKey string from an ID card."
+	desc = "A circuit that can read the registred name, assignment, and PassKey string from an ID card."
```
Comment (nwarz) [inline code/modules/integrated_electronics/subtypes/access.dm]: still a typo at "registred"

Comment (Alexch2) [reply code/modules/integrated_electronics/subtypes/access.dm]: 😂 🔫 
heck

**Pull request merged**